src/scheduler/node_info.c

```diff
@@ -3243,6 +3243,18 @@ is_vnode_eligible(node_info *node, resource_resv *resresv,
 		}
 	}
 
+	if (!dflt_sched) {
+		if (resresv->is_job && (node->partition != NULL)) {
+			if (cstrcmp(node->partition, resresv->job->queue->partition) != 0) {
+				set_schd_error_codes(err, NOT_RUN, INVALID_NODE_TYPE);
+				set_schd_error_arg(err, ARG1, (char*) node_state_to_str(node));
+#ifdef NAS /* localmod 031 */
+				set_schd_error_arg(err, ARG2, node->name);
+#endif /* localmod 031 */
+				return 0;
+			}
+		}
+	}
```

Comment:
Doing string comparisons in the scheduler is bad.  Doing them in node search is horrible.  You'll significantly hurt scheduler performance.  Create the list of nodes associated to a queue's partition on the queue itself.  This means we only are searching the right set of nodes.

Reply:
Implemented.